airbyte-integrations/connectors/source-square/source_square/source.py

```diff
@@ -358,16 +362,75 @@ def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
             yield {"location_ids": location}
 
 
+class Oauth2AuthenticatorSquare(Oauth2Authenticator):
+    def refresh_access_token(self) -> Tuple[str, int]:
+        """Handle differences in expiration attr:
+        from API: "expires_at": "2021-11-05T14:26:57Z"
+        expected: "expires_in": number of seconds
+        """
+        token, expires_at = super().refresh_access_token()
+        expires_in = pendulum.parse(expires_at) - pendulum.now()
+        return token, expires_in.seconds
+
+
 class SourceSquare(AbstractSource):
-    api_version = "2021-06-16"  # Latest Stable Release
+    api_version = "2021-09-15"  # Latest Stable Release
+
+    @staticmethod
+    def get_auth(config: Mapping[str, Any]) -> AuthBase:
+
+        credential = config.get("credentials", {})
+        auth_type = credential.get("auth_type")
+        if auth_type == "Oauth":
+            # scopes needed for all currently supported streams:
+            scopes = [
+                "CUSTOMERS_READ",
+                "EMPLOYEES_READ",
+                "ITEMS_READ",
+                "MERCHANT_PROFILE_READ",
+                "ORDERS_READ",
+                "PAYMENTS_READ",
+                "TIMECARDS_READ",
+                # OAuth Permissions:
+                # https://developer.squareup.com/docs/oauth-api/square-permissions
+                # https://developer.squareup.com/reference/square/enums/OAuthPermission
+                # "DISPUTES_READ",
+                # "GIFTCARDS_READ",
+                # "INVENTORY_READ",
+                # "INVOICES_READ",
+                # "TIMECARDS_SETTINGS_READ",
+                # "LOYALTY_READ",
+                # "ONLINE_STORE_SITE_READ",
+                # "ONLINE_STORE_SNIPPETS_READ",
+                # "SUBSCRIPTIONS_READ",
```

Comment:
Are we going to support these streams in the future?

btw ideally It seems like the spec.json could maybe expose some multi-checkbox option to select which streams will be synced (and should have permissions for) when configuring the source?
(which would adapt the requested scopes when oauthing)